mbq/atomiq/utils.py

```diff
@@ -27,3 +31,54 @@ def wrapped_func(*args, **kwargs):
 
         return wrapped_func
     return wrapper
+
+
+def has_user_transactions_in_django_test_case():
+    """
+    Atomiq publish will throw an exception if not used within a transaction, and we want
+    to make sure nobody ships code that is going to throw this exception in production.
+    The problem is that Django TestCase wraps all test functions in a transaction, which
+    masks this error. Therefore, we use this function to ~inspect~ ~the~ ~stack~ and look for
+    instances of django.test.TestCase. If we find TestCase unit test, then we expect there to be
+    TWO transactions created by Django, and ONE transaction created by the user.
+
+    Thanks for listening.
+    Your pal,
+
+    Per-Andre
+
+    """
+    in_setup = False
+    in_test_case = False
+
+    db_connection = transaction.get_connection()
+
+    # This loops through the call stack and sets "in_setup" and "in_test_case"
+    for stack_frame in inspect.stack():
+        for local_var in stack_frame[0].f_locals.values():
+            if stack_frame[3] in ['setUpTestData', 'setUpclass']:
+                in_setup = True
+
+            if not local_var:
+                continue
```

Comment:
Not sure I understand what this is doing, do the test functions below freak out if you call them on `None` or something?